### ISSUE TITLE
Any modeling of temperature and humidity etc?

### ISSUE THREAD

**Issue by sublimator** [mrc-ide/covid-sim#266]
https://www.google.com/search?q=wuhan+temperature+by+month
![image](https://user-images.githubusercontent.com/525211/82119010-2ab31980-97a5-11ea-9bc4-0da6f8c4c361.png)

https://www.youtube.com/watch?v=FI5MqXPSRD8
![image](https://user-images.githubusercontent.com/525211/82119017-43bbca80-97a5-11ea-97bb-c2dc6489f2ee.png)

https://papers.ssrn.com/sol3/Papers.cfm?abstract_id=3551767
![image](https://user-images.githubusercontent.com/525211/82119035-6a7a0100-97a5-11ea-8568-e2740df50036.png)

A quick grep of the code seemed to show know parameters for climate ?

**Comment by robscovell-ts** [comment]
There are two questions here:

1. Do temperature and humidity affect the virus particles themselves?
2. Do temperature and humidity affect human behaviour in ways that impact transmission?

I can't comment on the first question.

On the second question, I believe that there are some significant improvements to be made to the model around human social behaviour because those behaviours affect the viral load that members of a group of people will experience in the presence of infected hosts. Colder temperatures mean more people spend more time huddled inside with doors and windows closed, possibly with heat exchanger systems pumping virus-laden air around. Warmer temperatures mean more people outside. Temperatures change how people socialise, with steamy pubs replaced by beach barbecues. My understanding is that changes to the model based on these factors are not yet in scope due to the current work pressures being experienced by the ICL team.

**Comment by weshinsley** [comment]
This would be an interesting, challenging piece of work, but indeed, probably not for right now. I am not sure that there is sufficient data and understanding of the relationships between temperature/humidity and the virus.

The topic is quite old - look up papers by Jeffrey Shaman's group regarding flu some years ago. At that time, I don't think there was a convincingly strong direct connection between weather and virus transmission; possibly some indirect mechanisms (weather affecting people's behaviour, or even something as basic as school holidays in Summer causing reduced transmission).

At the moment, there is nothing in the code explicitly about weather conditions (and weather is the word you want I think, not climate...) - but you can set school holidays, and also seasonality parameters can be set to model variations through the calendar year - but I think it's not yet clear with covid what that parameterisation should be.

**Comment by sublimator** [comment]
> these factors are not yet in scope 

My intuition is that it's a **very** important factor. I personally wouldn't trust or use any model that didn't account for it.

I say that because I live in Cambodia and my experience here. In the early days of the outbreak the prime minister here [travelled to China](https://asia.nikkei.com/Politics/International-relations/A-friend-in-need-is-a-friend-indeed-Xi-tells-visiting-Hun-Sen) as a sign of solidarity. The general line from the Ministry of Health was that there was nothing to worry about in Cambodia due to the warm weather. "Fear is more dangerous than the virus itself".  They said they would not shut down any travel from China and never have.

In fact Cambodia [accepted a cruise ship](https://www.inquirer.com/news/nation-world/coronavirus-cruise-ship-stranded-cambodia-20200214.html
) that was stranded around valentines day:
![image](https://user-images.githubusercontent.com/525211/82354108-9659e880-9a2a-11ea-841c-8416900996bd.png)

This was before EU and USA were infected and people were generally happy with everything. 

We have only had 122 officially reported cases, no new ones for the last month, and our last active case cleared up a few days ago. Zero deaths. I (and no one I know of here) have *any* reason to believe there is some coverup going on. (Of course I'm under no illusion those numbers *exactly* match reality. )

Note that it's *really* quite balmy here:
![image](https://user-images.githubusercontent.com/525211/82353280-678f4280-9a29-11ea-976e-946beb1439f0.png)

Life is starting to return to normal:
![image](https://user-images.githubusercontent.com/525211/82354735-69f29c00-9a2b-11ea-9cf9-d492d76117dd.png)

Related to weather is of course vitamin D but how long the virus can last on surfaces could be HUGE. There is some reason to believe/intuit exposure to less "load" of the virus would yield a less brutal infection. It would seem in cold climates where a lot of people get sick and you are constantly exposed ...

**Comment by sublimator** [comment]
And somewhat semi seriously, [space weather](https://cosmictusk.com/wp-content/uploads/CurrentScience2020-copy.pdf) ?

**Comment by bbolker** [comment]
It is really, really hard to sort out effects of climate, culture, intervention, exposure, etc..

Some starting points in the literature:

- https://science.sciencemag.org/content/early/2020/05/15/science.abc2535
- https://www.thelancet.com/journals/lanplh/article/PIIS2542-5196(20)30106-6/fulltext
- https://www.medrxiv.org/content/10.1101/2020.04.14.20064923v1

>  Conclusion: Considering the existing scientific evidence, warm and wet climates seem to reduce the spread of COVID-19. The certainty of the evidence generated was graded as low. However, these variables alone could not explain most of the variability in disease transmission.

**Comment by sublimator** [comment]
Models are worthless then :(

**Comment by robscovell-ts** [comment]
@sublimator --
All models are limited by the following:

* identification of relevant factors
* ability to parameterise relevant factors
* possibility of incorrect or untested assumptions
* unconscious bias towards or away from some factors, leading to weighting issues.

When you run your model, it is important to retain awareness of the limitations. This model makes use of population density data and a single probability of infection from individual interactions. It can therefore tell you by how much you need to artificially reduce population density through social distancing and lockdown measures. It can't help you with risk analysis around different social scenarios, e.g. pubs, churches, weddings, conventions, workplaces etc. To undertake that kind of risk analysis requires a different kind of model for movement of viral particles in enclosed spaces over time.

Because Cambodia is hot and humid, much of life is outside in the fields, in the street, or in houses with open sides, allowing virus particles to disperse more quickly.

**Comment by sublimator** [comment]
Does this model social clusterings? People are, after all, creatures of
habit, not ants wandering cells with random interactions.

Does the virus not have some kind of outer lipid shell? Susceptible to heat?

**Comment by robscovell-ts** [comment]
@sublimator I think it doesn't take a model to realise that social clustering will increase infection rates, so on the upside of the sombrero eliminating social clustering was necessary. Now we seem to be on the down side, I think it is time to bring in other models that allow for risk assessment for different scenarios of social clustering, so we're not stuck with a stark binary choice between 'health' and 'economy'. However, this is very hard to do because there are so many unknowns, and a generalised model is not going to fit every possible real-life scenario, so expect to see confused messaging from policy-makers as they try to reconcile contradictions. I think then it does come down to common sense in the specific situations you find yourself in. You'll probably be OK at a beach barbecue but snogging random pretty strangers in a steamy pub is probably not such a good idea. There comes a point at which you need to take responsibility for your own decisions and risk assessment and not look to the government for advice before you do anything!

**Comment by bbolker** [comment]
@sublimator: I think this issues list is not the best forum for **general** discussion of COVID modelling problems (see also my comment [here](https://github.com/mrc-ide/covid-sim/issues/247#issuecomment-628282774)). If you have *specific* suggestions or questions about how to implement code to incorporate the effects of temperature and humidity in the simulation (preferably along with suggestions for how to estimate/calibrate the new parameters!), that's great (even better if you go ahead and implement your suggestions in a well-tested fork/pull request that can be shown to (1) not break the existing code and (2) maintain backward compatibility). Unfortunately, general comments along the lines of "you guys ought to incorporate XXX" (or "your model sucks", or "all models suck") are likely to be ignored.

**Comment by robscovell-ts** [comment]
@bbolker I have to disagree I'm afraid, sorry. The decision has been made to open up the model to public scrutiny and the result has been to allow press hacks to do a hatchet job on the model. The model (among others) has been used as an important driver of public policy. This forum is one of the few places where public scrutiny can take place. The public are the ultimate 'client' here: public money funds the modelling process, which then impacts on the public in the form of policy. The public are entitled to scrutinise the model they have paid for without the scrutiny being filtered through 'representatives', or 'communications managers'. 

This forum allows members of the public to understand the limitations of the model, and of modelling generally, so plays a very important educational role. It's a valuable window into the mathematical modelling community, and it's one that I very much welcome, because what we do is usually surrounded by an aura of mystery onto which people can impose either their prejudices or their unwonted admiration. 

I came here thinking (from news reports) that the team needed help with the software engineering aspect of their work, and I discovered that the news reports were wrong. I think that's a valuable outcome in itself.

**Comment by bbolker** [comment]
Your points are well taken.  I have two concerns:

- for better or worse the modelling team are too busy to spend very much time engaging with this forum (you can argue that they should, but the people who are making the decision about allocation of effort to public engagement are (I suspect) above the pay grade of the people actually maintaining this repository. The net effect is that people like you and I, who are **not** the ones with detailed knowledge of the code and the history of modelling decisions, end up being the ones responding.
- this issue thread isn't too bad, but a number of the threads here have attracted nasty and politically motivated commenters. Maybe that's the price to pay for public engagement, but again I think it's not what the maintainers of this repository had in mind (nor what they're being paid to deal with ...)

**Comment by robscovell-ts** [comment]
Your points are well taken too, @bbolker.

**Comment by robscovell-ts** [comment]
I think people like you and I have a role in helping to educate people who come on here to comment, within the bounds of our knowledge of this model and of modelling in general.

**Comment by bbolker** [comment]
@sublimator: I'm not entirely sure I understand your comments, but I'll try.

> Does this model social clusterings? People are, after all, creatures of
habit, not ants wandering cells with random interactions.

As described in the [overview](https://github.com/mrc-ide/covid-sim/blob/master/docs/model-overview.md), the model includes structure at the level of spatial cells; spatial microcells; 'households', 'places' (and 'place groups' within places) and models the probability of contact with people in another place/region/etc..  In more detail from [here](https://www.imperial.ac.uk/media/imperial-college/medicine/sph/ide/gida-fellowships/Imperial-College-COVID19-NPI-modelling-16-03-2020.pdf):

> Contacts with other individuals in the population are made within the household, at school, in the workplace and in the wider community ... Data on the distribution of
workplace size was used to generate workplaces with commuting distance data used to locate workplaces appropriately across the population. Individuals are assigned to each of these locations at the start of the simulation. ... Transmission events occur through contacts made between susceptible and infectious individuals in either the household, workplace, school or randomly in the community, with the latter depending on spatial distance between contacts.

The model doesn't currently include [dynamic mixing, e.g. at sporting events or concerts](https://github.com/mrc-ide/covid-sim/issues/210).

> Does the virus not have some kind of outer lipid shell? Susceptible to heat?

I don't know what the question is here.  There are experimental (lab) studies on viability of SARS-COV-2 under different environmental conditions (e.g. [here](https://www.hindawi.com/journals/av/2011/734690/)) but @robscovell-ts suggests it's hard to get all the detail we need on the microenvironments in which transmission occurs (temperature, humidity, UV, air movement ...) that would allow us to confidently translate these lab results to parameters of population spread.

**Comment by sublimator** [comment]
> I'm not entirely sure I understand your comments, but I'll try.

Sorry, a late reply, sent while sleepy and disappointed. I recall friends sending me quite crude simulations before that ran in browsers and noting the ant like behavior.

> Does the virus not have some kind of outer lipid shell? Susceptible to heat?

I mean the heat **seems** to cause **more than just behavioral changes** in humans. 

> Unfortunately, general comments along the lines of "you guys ought to incorporate XXX"

I don't mean to be to offensive l but I do recall **this particular model** mentioned in the media quite often over the last months and it seeming to be **quite influential and having quite an impact**. 

I watched **VN put in very strong measures** (early Feb) while many of us in **Cambodia shook our heads** for months. They */did/* close some schools/businesses, seal boarders etc eventually but not after months of unrestricted flights from China and around the world. 

And in the end ? What did all those extra measures get VN? Granted they are opening their schools. The Ministry of Health requested the schools to be opened here 3-4 weeks ago but the department of education decided not to.

> have attracted nasty and politically motivated commenters

Unfortunately the idea that this could be effected strongly by weather seems to elicit "you idiot, you listen to Trump" 

But what if? **What if the temperature */does/* play a huge role in this ?**

I'm sure people can find better quality data, but an image a friend posted a few months back is quite telling:
![image](https://user-images.githubusercontent.com/525211/82405702-d94da780-9a8e-11ea-893d-f7b8c0a86528.png)

> **much of life** is outside in the fields, in the street, or in houses with open sides, **allowing virus particles to disperse more quickly.**

That does seem to imply something against some of the more stringent lockdowns seen around the world. 

@robscovell-ts 
> When you run your model, it is important to retain awareness of the limitations.
Thanks for the info
Fair, I'll stick with my mental model for the moment :) 

@bbolker 
Thanks also!

**Comment by sublimator** [comment]
@weshinsley 
Thanks for the response

> weather is the word you want I think, not climate

Yes and no :) The definition of climate:

> the weather conditions prevailing in an area in general or over a long period.
> "our cold, wet climate"
> 

But if you could model fluctuating temperature changes due to weather (even day and night) that would be nice.

**Comment by robscovell-ts** [comment]
@sublimator I agree that the "Trump said it so it must be wrong" approach is getting a bit old. I think Trump enjoys riling people up for the fun of it, to be honest. Don't take him too seriously either way. 

I do remember there being maps showing correlation between latitude and coronavirus outbreaks early on. They made me go 'hmmmmm'. But always remember the dictum about correlation and causation -- it might be related to similar human behaviour at similar latitudes or it might just be coincidence. I would need more data on infection rates at different latitudes to work on that one.

Mental models are useful for developing personal heuristics based on personal observation. I recommend you read Nicholas Nassim Taleb. He is controversial in the modelling community, though, I warn you, and I haven't yet made up my mind about him. He is very good on common sense as a distillation and filtration of centuries of combined mental models. 

As a modeller you *have* to be open minded. Otherwise you're prone to getting attached only to those models that suit your biases.

**Comment by sublimator** [comment]
@robscovell-ts

> correlation and causation
Right

> I recommend you read Nicholas Nassim Taleb. 
> He is controversial in the modelling community

Cheers! I'm a programmer but not a modeler. As a single Dad I am barely getting work done lately, let alone have any time for amateur modeling. Some reading on the subject will be nice though, ... one day!

I have to say the Ministry of Health here pretty much called it in January IMO. I hope more research on the topic makes its way into these models given their impact.

Cheers!

**Comment by NeilFerguson** [comment]
An interesting discussion. We have distantly related code which models malaria, where a detailed dynamical representation of climate is included. In theory it would not be difficult to add to this code, but parameterising the relationships is better done with simpler models. This paper - https://science.sciencemag.org/content/368/6493/860.full - shows some evidence of seasonality in endemic human coronaviruses. That type of sinusoidal seasonality is already included in the code, but we haven’t used it as yet.

**Comment by bbolker** [comment]
Also: https://projects.iq.harvard.edu/covid19 (working paper [here](https://projects.iq.harvard.edu/files/covid19/files/weather_and_covid-19_preprint.pdf) might provide useful information for someone who wants to tackle this.

> We project the CRW [Relative COVID-19 Risk due to Weather] for the next year across the world using weather data (averaged over a moving window of 15 days) from 2019 period. Our projections suggest warmer times of the year, and locations, may offer a modest reduction in reproductive number, helping with efforts to contain the pandemic and build response capacity. Outdoor UV is also associated with lower transmission up to a point (index of ~6.5), but above the effect reverses, creating an overall U-shaped relationship. Pressure, precipitation, diurnal temperature, lack of humidity, SO2 and Ozone all modestly contribute to increased transmission rates. Overall, CRW numbers rarely go below 0.5 or above 1.5 [i.e. a 50% decrease or increase in transmissibility], indicating that upcoming changes in weather alone will not be enough to fully contain the transmission of COVID-19.

**Comment by sublimator** [comment]
I note this paper has an update recently:
https://papers.ssrn.com/sol3/Papers.cfm?abstract_id=3551767
